### PR TITLE
[#1690] improvement(core): Decode byte array value of entity in log to make it more readable

### PR DESCRIPTION
**Pull request by yuqi1129**
### What changes were proposed in this pull request?

Decode the byte array into an entity and its corresponding type and creation time.

### Why are the changes needed?

To make logs easier to read. 

Before:
```text

Physically delete key that has newer version: ml/ _\xC00\xAF?\x99\xA3\x1F\xF9\xCB\x14?W\x17\xE6\x0B, a newer version ml/ _\xC00\xAF?\x99\xA3\x1F\xF9\xCB\x14=0\xBB\xA4\xA9
```

after:
```text
name identity: 'test_2019', entity type: 'METALAKE', createTime: '2024-01-24 21:27:50(1706102870533)', newVersion createTime: '2024-01-24 21:28:04(1706102884326)', key: 'ml/\x06\xC6\x1E\xD7\x85Z\xC3\xC3\x1F\xF9\xCB\x11`\xB7\xEB\xFE\x0C', newVersion key: 'ml/\x06\xC6\x1E\xD7\x85Z\xC3\xC3\x1F\xF9\xCB\x11_\xE0g\xFC\xE0'
```

image

Fix: #1690

### Does this PR introduce _any_ user-facing change?

N/A

### How was this patch tested?

Not yet.

### CODE REVIEW

**Comment by yuqi1129** [comment]
Blocked by #1664

File: core/src/main/java/com/datastrato/gravitino/storage/kv/KvNameMappingService.java [hunk 1]
```diff
@@ -73,7 +73,7 @@ public String getNameById(long id) throws IOException {
             FunctionUtils.executeInTransaction(
                 () -> {
                   byte[] name = transactionalKvBackend.get(idByte);
-                  return idByte == null ? null : new String(name);
```
Comment (jerryshao) [inline core/src/main/java/com/datastrato/gravitino/storage/kv/KvNameMappingService.java:76]: Is it a bug in previous code?

Comment (yuqi1129) [reply core/src/main/java/com/datastrato/gravitino/storage/kv/KvNameMappingService.java:76]: yes

File: core/src/main/java/com/datastrato/gravitino/storage/kv/KvGarbageCollector.java [hunk 2]
```diff
@@ -156,7 +175,14 @@ private void collectAndRemoveOldVersionData() throws IOException {
 
         // Value has deleted mark, we can remove it.
         if (null == TransactionalKvBackendImpl.getRealValue(rawValue)) {
-          LOG.info("Physically delete key that has marked deleted: {}", Bytes.wrap(key));
+          LogHelper logHelper = decodeKey(key, transactionId);
+          LOG.info(
+              "Physically delete key that has marked deleted:name identity: '{}', entity type: '{}', createTime: '{}({})', key: '{}'",
```
Comment (jerryshao) [inline core/src/main/java/com/datastrato/gravitino/storage/kv/KvGarbageCollector.java:180]: -> ": name identifier: "

File: core/src/main/java/com/datastrato/gravitino/storage/kv/KvGarbageCollector.java [hunk 3]
```diff
@@ -176,23 +202,92 @@ private void collectAndRemoveOldVersionData() throws IOException {
                     .build());
         if (!newVersionOfKey.isEmpty()) {
           // Has a newer version, we can remove it.
+          LogHelper logHelper = decodeKey(key, transactionId);
+          byte[] newVersionKey = newVersionOfKey.get(0).getKey();
+          LogHelper newVersionLogHelper = decodeKey(newVersionKey);
           LOG.info(
-              "Physically delete key that has newer version: {}, a newer version {}",
+              "Physically delete key that has newer version: name identity: '{}', entity type: '{}', createTime: '{}({})', newVersion createTime: '{}({})',"
```
Comment (jerryshao) [inline core/src/main/java/com/datastrato/gravitino/storage/kv/KvGarbageCollector.java:209]: Also here.

File: core/src/main/java/com/datastrato/gravitino/storage/kv/KvGarbageCollector.java (hunk 2 above)
Comment (yuqi1129) [reply core/src/main/java/com/datastrato/gravitino/storage/kv/KvGarbageCollector.java]: Fixed

**Review by jerryshao — APPROVED** [review]

**Pull request merged**